src/amo/sagas/collections.js

```diff
@@ -0,0 +1,48 @@
+import { all, call, put, select, takeEvery } from 'redux-saga/effects';
+import {
+  FETCH_COLLECTION,
+  loadCollection,
+} from 'amo/reducers/collections';
+import * as api from 'amo/api/collections';
+import log from 'core/logger';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+export function* fetchCollection({
+  payload: {
+    errorHandlerId,
+    page,
+    slug,
+    user,
+  },
+}) {
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const state = yield select(getState);
+
+    const { detail, addons } = yield all({
+      detail: call(api.getCollectionDetail, {
+        api: state.api,
+        slug,
+        user,
+      }),
+      addons: call(api.getCollectionAddons, {
+        api: state.api,
+        page,
+        slug,
+        user,
+      }),
+    });
+
+    yield put(loadCollection({ addons, detail }));
+  } catch (error) {
+    log.warn(`Collection failed to load: ${error}`);
+    yield put(errorHandler.createErrorAction(error));
+  }
+}
+
+export default function* collectionsSaga() {
+  yield takeEvery(FETCH_COLLECTION, fetchCollection);
```

Comment:
I think we should default to `takeLatest` unless we know that we need to process parallel requests.